### ISSUE TITLE
Unknown layer type: ImageData

### ISSUE THREAD

**Issue by rickPPD** [facebook/C3D#348]
I build a docker with bvlc/caffe:gpu, and compile C3D-v1.1

when I run the demo feature_extraction at:
`./build/tools/extract_features.bin models/bvlc_reference_caffenet/bvlc_reference_caffenet.caffemodel examples/_temp/imagenet_val.prototxt fc7 examples/_temp/features 10 leveldb`

and I got the error

E0103 13:51:56.192770    12 extract_features.cpp:62] Using CPU
F0103 13:51:56.345430    12 layer_factory.hpp:81] Check failed: registry.count(type) == 1 (0 vs. 1) Unknown layer type: ImageData (known types: AbsVal, Accuracy, ArgMax, BNLL, BatchNorm, BatchReindex, Bias, BlackHole, Concat, ContrastiveLoss, Convolution, Convolution3D, Crop, Data, Deconvolution, Deconvolution3D, Dropout, DummyData, ELU, Eltwise, Embed, EuclideanLoss, Exp, Filter, Flatten, HDF5Data, HDF5Output, HingeLoss, Im2col, InfogainLoss, InnerProduct, Input, L2Normalize, LRN, LSTM, LSTMUnit, Log, MVN, MemoryData, MultinomialLogisticLoss, PReLU, Parameter, Pooling, Pooling3D, Power, Python, RNN, ReLU, Reduction, Reshape, SPP, Scale, Sigmoid, SigmoidCrossEntropyLoss, Silence, Slice, Softmax, SoftmaxWithLoss, Split, TanH, Threshold, Tile)

so what`s wrong here? 
Thanks.

**Comment by dutran** [comment]
I think the prototxt you use is with a later version of caffe, while this branch is earlier where there is no ImageData layer.

**Comment by vishalkumarchaudhary** [comment]
uncomment ``` OPENCV :=3 ``` in Makefile.config then 
```
sudo make clean |  sudo make all
```
